pootle/apps/pootle_translationproject/views.py

```diff
@@ -50,6 +50,51 @@ def admin_permissions(request, translation_project):
                        'translation_projects/admin/permissions.html', ctx)
 
 
+def get_sidebar_announcements_context(request, project, language):
+    announcements = []
+    new_cookie_data = {}
+    cookie_data = {}
+
+    if SIDEBAR_COOKIE_NAME in request.COOKIES:
+        json_str = unquote(request.COOKIES[SIDEBAR_COOKIE_NAME])
+        cookie_data = json.loads(json_str)
+
+    is_sidebar_open = cookie_data.get('isOpen', True)
+
+    def get_announcement(virtual_path):
+        try:
+            announcement = StaticPage.objects.live(request.user).get(
+                virtual_path=virtual_path,
+            )
+        except StaticPage.DoesNotExist:
+            pass
+        else:
+            announcements.append(announcement)
+
+            ann_mtime = dateformat.format(announcement.modified_on, 'U')
+            stored_mtime = cookie_data.get(virtual_path, None)
+
+            if ann_mtime != stored_mtime:
+                is_sidebar_open = True
+                new_cookie_data[virtual_path] = ann_mtime
+
+    get_announcement('announcements/projects/' + project.code)
```

Comment:
Renamed the function to `_set_announcement_context_and_cookie` so it does what its name says. Also added a docstring which explains the code a bit more.

Changed to use interpolation. If with "build the string internally" you mean having a if..elif..else to build the virtual path, I disagree.